src/std/misc/string.ss

```diff
@@ -77,3 +79,87 @@ package: std/misc
     ((_ eol fallback) (let ((trimmed (string-trim-suffix eol string)))
                         (if (eq? trimmed string) fallback (values trimmed eol)))))
   (try +crlf+ (try +lf+ (try +cr+ (values string "")))))
+
+
+;; Returns a single character-string of s at the index i
+;; or an error if the index could not be found.
+(def (string-at s i)
+  (string (string-ref s i)))
+
+
+;; string-subst helper which handels the case
+;; that the argument 'old' is an empty string.
+;;   new    non-empty
+;;   limit  non-zero
+(def (subst-helper-empty-old str new limit)
+  (declare (fixnum))
+  (def len-str (string-length str))
+  (if (= limit 1)
+    (string-append new str)
+    (let ((stop (1- (string-length str)))
+          (limit (if (or (negative? limit)
+                         (> limit len-str)) (1+ len-str) limit)))
+      (string-concatenate
```

Comment:
Instead of `string-concatenate`, shouldn't we be using a `call-with-output-string` ?
Then we don't need `string-at`, we can just `write-char` the character to the stream.

Reply:
Very good idea, thanks!